client/settings/check-starcraft-path.js

```diff
@@ -42,14 +58,22 @@ async function checkHash(path, validHashes) {
 // be checked for other copies. If downgradePath contains files that match the correct hashes, this
 // will be counted as having the correct version, but `downgradePath` will be true.
 export async function checkStarcraftPath(dirPath, downgradePath) {
-  const requiredFiles = ['starcraft.exe', 'storm.dll', 'local.dll', 'stardat.mpq', 'broodat.mpq']
+  const requiredFiles = ['starcraft.exe', 'storm.dll', 'stardat.mpq', 'broodat.mpq']
 
   try {
     await Promise.all(requiredFiles.map(f => accessAsync(path.join(dirPath, f), fs.constants.R_OK)))
   } catch (err) {
     return { path: false, version: false, downgradePath: false }
   }
 
+  // Due to 1.19 version moving local.dll to a separate folder, we need to handle it separately
+  const matches = await globAsync(`${dirPath}/**/local.dll`)
+  if (matches.length > 0) {
+    maybeCopyLocalDll(matches[0], downgradePath)
```

Comment:
We don't do anything for 1.16.1 installs, we should continue not doing anything for those. I meant that you should copy it *any time we downgrade*, not always. And you definitely *should not* be copying as part of something that is just meant to check whether an install is valid